core/src/main/java/hudson/model/TreeView.java

```diff
@@ -140,9 +140,16 @@ public void deleteView(View view) throws IOException {
     }
 
     public View getView(String name) {
-        for (View v : views)
-            if(v.getViewName().equals(name))
+        for (View v : views) {
+            if (v.getViewName().equals(name))
                 return v;
+            if (v instanceof ViewGroup) {
```

Comment:
The only problem with this approach is that Nested views may appear as a search result even if there is TopLevel view with such name. And as a user I would expect them to have higher priority. Maybe it worth checking all TopLevel views first.

I would also vote for making a parameterizable method `View getView(String name, boolean recursive)` and maybe changing only the API calls which are really related to search. Otherwise it it hard to predict the impact of the defaukt behavior change

Reply:
> And as a user I would expect them to have higher priority.

FWIW, I checked that the selection takes you to the top level view, so I think the current behavior is fine.